cli/index_data.py

```diff
@@ -96,6 +123,47 @@ def get_document_generator(
             }
 
 
+def populate_and_warmup_index(
+    doc_generator: Generator[dict, None, None], index_name: str
+):
+    """
+    Load documents into an Opensearch index and load the KNN index into native memory (warmup).
+
+    :param doc_generator: generator of Opensearch documents to index
+    :param index_name: name of index to load documents into
+    """
+
+    logger.info(f"Loading documents into index {index_name}")
+
+    opensearch = OpenSearchIndex(
```

Comment:
Would it be better to add this as an argument to the function so as to group environment variable reads from a main  file rather than the function itself?

Reply:
The issue with adding it as a function argument is that at the moment `OpenSearchIndex` is defined per *index*, meaning we'd need to create an `OpenSearchIndex` instance for each call to `populate_and_warmup_index()` in `main()`.